com.unity.render-pipelines.universal/Shaders/Utils/TileDepthInfo.shader

```diff
@@ -35,6 +35,277 @@ Shader "Hidden/Universal Render Pipeline/TileDepthInfo"
+    #endif
+    TEXTURE2D_FLOAT(_DepthTex);
+    float4 _DepthTexSize;
+    // Missing C# interface to upload int vectors!
+    int _DownsamplingWidth;
+    int _DownsamplingHeight;
+    int _tileXCount;
+    int _DepthRangeOffset;
+    int _SourceShiftX;
+    int _SourceShiftY;
+    int _TileShiftX;
+    int _TileShiftY;
+    float4 _unproject0;
+    float4 _unproject1;
+
+    uint GenerateBitmask(float4 ds, half listDepthFactorA, half listDepthFactorB)
+    {
+        #if UNITY_REVERSED_Z // TODO: can fold reversed_z into g_unproject parameters.
+        ds = 1.0 - ds;
+        #endif
+
+        // View space depth (absolute).
+        half4 geoDepths = -(_unproject0.z * ds + _unproject0.w) / (_unproject1.z * ds + _unproject1.w);
+        int4 bitIndices = geoDepths * listDepthFactorA + listDepthFactorB;
+        // bitshift operator has undefined behaviour if bit operarand is out of range on some platforms.
+        // On Switch, keeping the test slightly fasten shader execution.
+        bool4 isValid = (bitIndices >= 0 && bitIndices <= 31); 
+        uint4 bits = isValid << bitIndices;
+        return bits.x | bits.y | bits.z | bits.w;
+    }
+
+    uint fragBitmaskDownsample(Varyings input) : SV_Target
+    {
+        UNITY_SETUP_STEREO_EYE_INDEX_POST_VERTEX(input);
+        UNITY_SETUP_INSTANCE_ID(input);
+
+        #if defined(DOWNSAMPLING_SIZE_2)
+            #define DOWNSAMPLING_SIZE 2
+        #elif defined(DOWNSAMPLING_SIZE_4)
+            #define DOWNSAMPLING_SIZE 4
+        #elif defined(DOWNSAMPLING_SIZE_8)
+            #define DOWNSAMPLING_SIZE 8
+        #elif defined(DOWNSAMPLING_SIZE_16)
+            #define DOWNSAMPLING_SIZE 16
+        #endif
+
+        #if defined(DOWNSAMPLING_SIZE)
+            const int2 downsamplingSize = int2(DOWNSAMPLING_SIZE, DOWNSAMPLING_SIZE);
+            #define DOWNSAMPLING_LOOP [unroll]
+        #else
+            const int2 downsamplingSize = int2(_DownsamplingWidth, _DownsamplingHeight);
+            #define DOWNSAMPLING_LOOP [loop]
+        #endif
+
+        int2 fragCoord = int2(input.positionCS.xy);
+        int2 sourceCorner = (fragCoord << int2(_SourceShiftX, _SourceShiftY));
+        int2 tileCoord = fragCoord >> int2(_TileShiftX, _TileShiftY);
+        uint tileID = tileCoord.x + tileCoord.y * _tileXCount;
+        uint listDepthRange = LoadDepthRange(tileID - _DepthRangeOffset);
+        half listDepthFactorA = f16tof32(listDepthRange);       // a = 32.0 / (listMaxDepth - listMinDepth)
+        half listDepthFactorB = f16tof32(listDepthRange >> 16); // b = 32.0 / (listMaxDepth - listMinDepth)
+
+        uint geoBitmask = 0;
+
+        #if !USE_GATHER // Base reference version.
+            DOWNSAMPLING_LOOP for (int j = 0; j < downsamplingSize.y; ++j)
+            DOWNSAMPLING_LOOP for (int i = 0; i < downsamplingSize.x; ++i)
+            {
+                int2 tx = sourceCorner + int2(i, j);
+                tx.y = _DepthTexSize.y - tx.y; // TODO fixme: Depth buffer is y-inverted
```

Comment:
does this depend on API? gl uv(0,0) == left-bottom

Reply:
It could, although for the purpose of downsizing the depth buffer, it should not matter. GL is also not officially supported for deferred renderer. I think the comment above is referring to URP y-flipping the rendering, which causes issues later on as the logic for numbering tile is (0,0) for top-left corner.